### PR TITLE
APM-561 Update payment endpoint

### PR DESCRIPTION
**Pull request by ndijkstra**

### CODE REVIEW

**Comment by sandervanhooft** [comment]
This is great. I’ll try to find some time in the coming work days to do a PR for the php client.

**Comment by ndijkstra** [comment]
It's not yet official 👀 so there can still change some things

File: source/changelog/v2/changelog.rst [hunk 1]
```diff
@@ -6,6 +6,11 @@ changes are documented here.
 August 2019
 ===========
 
+Monday, 26th
+------------
+
+- Introduced the new endpoint to update some details of created payments. See :doc:`Update payment </reference/v2/payments-api/update-payment>`
```
Comment (shirshir) [inline source/changelog/v2/changelog.rst]: `Introduced a new endpoint...` 
`See ... for more info.`

File: source/reference/v2/payments-api/update-payment.rst [hunk 2]
```diff
@@ -0,0 +1,163 @@
+Update payment
+==============
+.. api-name:: Payments API
+   :version: 2
+
+.. endpoint::
+   :method: PATCH
```
Comment (shirshir) [inline source/reference/v2/payments-api/update-payment.rst]: Also `POST`?

File: source/reference/v2/payments-api/update-payment.rst [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+   :url: https://api.mollie.com/v2/payments/*id*
+
+.. authentication::
+   :api_keys: true
+   :organization_access_tokens: true
+   :oauth: true
+
+This endpoint can be used to update some details of a created payment.
```
Comment (shirshir) [inline source/reference/v2/payments-api/update-payment.rst]: I guess this should also mention when this endpoint can be used (payment should be `open`).

File: source/reference/v2/payments-api/update-payment.rst [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+          :required: false
+
+     - The date the payment should :doc:`expire </payments/status-changes>`, in ``YYYY-MM-DD`` format.
+       **Please note:** the minimum date is tomorrow and the maximum date is 100 days after tomorrow.
+
+       **Updating this field is only possible when the payment is not yet finalized.**
+
+Response
+--------
+``201`` ``application/hal+json``
```
Comment (shirshir) [inline source/reference/v2/payments-api/update-payment.rst]: Should be `200`.

File: source/reference/v2/payments-api/update-payment.rst (hunk 2 above)
Comment (shirshir) [reply source/reference/v2/payments-api/update-payment.rst]: Hmm, nevermind, it's also not mentioned in the "update order" endpoint.

File: source/reference/v2/payments-api/update-payment.rst (hunk 3 above)
Comment (shirshir) [reply source/reference/v2/payments-api/update-payment.rst]: Or `not finalized`.

Comment (ndijkstra) [reply source/reference/v2/payments-api/update-payment.rst]: For some fields it is possible to update them even when finalized

Comment (shirshir) [reply source/reference/v2/payments-api/update-payment.rst]: Ah, ok 👍

**Pull request merged**